core/shared/src/main/scala/scalaz/zio/ZSchedule.scala

```diff
@@ -197,7 +203,7 @@ trait ZSchedule[-R, -A, +B] extends Serializable { self =>
    * continue, using the maximum of the delays of the two schedules.
    */
   final def &&[R1 <: R, A1 <: A, C](that: ZSchedule[R1, A1, C]): ZSchedule[R1, A1, (B, C)] =
-    combineWith(that)(_ && _, _ max _)
+    combineWith(that)(_ && _, (a, b) => Max(a, b))
```

Comment:
I'd add `max` and `min` functions to `Delay` so you don't have to change this code.